backend/src/secondary-adapters/middleware/authentication-middleware.ts

```diff
@@ -0,0 +1,40 @@
+import express from 'express';
+import { ExtendedRequest } from '~/shared/types/express';
+import { jwtVerify } from 'jose';
+import { createSecretKey } from 'node:crypto';
+import { CONFIG } from '~/configuration/config';
+import { UserBaseResponseDto } from '~/shared/types/types';
+import { Unauthorized } from '~/shared/exceptions/unauthorized';
+import { exceptionMessages } from '~/shared/enums/exceptions';
+
+export const authenticationMiddleware =
+  (isAuthRequred = true) =>
+  async (req: ExtendedRequest, _res: express.Response, next: express.NextFunction): Promise<void> => {
+    req.user = null;
```

Comment:
Looks like it's not needed. Or explain please

Reply:
`isAuthRequired` is a feature for endpoints who could utilize the info about current user, but not necessarily require it and can continue even without authentication (in this case `req.user` will be `null`). there is currently no use case for it, but it could arise in the future, and even if it doesn't this feature doesn't add any cost to using this middleware, so I would prefer to let it stay.